flair/models/sequence_tagger_model.py

```diff
@@ -941,6 +942,96 @@ def _fetch_model(model_name) -> str:
 
         return model_path
 
+    def _generate_model_card(self, repo_id):
+        return f"""---
+tags:
+- flair
+- token-classification
+- sequence-tagger-model
+---
+
+### Demo: How to use in Flair
+
+Requires:
+- **[Flair](https://github.com/flairNLP/flair/)** (`pip install flair`)
+
+```python
+from flair.data import Sentence
+from flair.models import SequenceTagger
+
+# load tagger
+tagger = SequenceTagger.load("{repo_id}")
+
+# make example sentence
+sentence = Sentence("On September 1st George won 1 dollar while watching Game of Thrones.")
+
+# predict NER tags
+tagger.predict(sentence)
+
+# print sentence
+print(sentence)
+
+# print predicted NER spans
+print('The following NER tags are found:')
+
+# iterate over entities and print
+for entity in sentence.get_spans('ner'):
+    print(entity)
+```"""
+
+    def push_to_hub(
+        self,
+        repo_id: str,
+        token: Optional[str] = None,
+        private: bool = None,
+        commit_message: str = "Add new SequenceTagger model.",
+    ):
+        """
+        Uploads the Sequence Tagger model to a Hugging Face Hub repository.
+        :param repo_id: A namespace (user or an organization) and a repo name separated by a `/`.
+        :param token: An authentication token (See https://huggingface.co/settings/token).
+        :param private: Whether the repository is private.
+        :param commit_message: Message to commit while pushing.
+        :return: The url of the repository.
+        """
+        # Lazy import
+        from huggingface_hub import create_repo, model_info, upload_folder
+
+        repo_url = create_repo(
+            repo_id=repo_id,
+            token=token,
+            private=private,
+            exist_ok=True,
+        )
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            tmp_path = Path(tmp_dir)
+
+            # Save model weight
+            local_model_path = tmp_path / "pytorch_model.bin"
+            self.save(local_model_path)
+
+            # Determine if model card already exists
+            info = model_info(repo_id)
```

Comment:
@alanakbik Thanks for spotting this issue. I have been able to reproduce it myself. The problem comes from this line where we try to fetch info from the model (to check if readme already exists or not). Token is not provided to `model_info` which makes it fail on private repos.

@osanseviero You just have to update this line as below and it works fine (I tested the fix locally):
```suggestion
            info = model_info(repo_id, use_auth_token=token)
```